pkg/framework/aries/api/vdr/vdr.go

```diff
@@ -98,3 +99,10 @@ func WithOption(name string, value interface{}) DIDMethodOption {
 		didMethodOpts.Values[name] = value
 	}
 }
+
+// WithKeyType adds key type option for did method.
+func WithKeyType(kt kms.KeyType) DIDMethodOption {
```

Comment:
please remove this option and use WithOption need to keep VDR interface generic

Reply:
I think this option should be removed altogether to make this interface less confusing.

The method already accepts a `*did.Doc`. I'm already using _that_ to set the verification methods I need in downstream projects. I don't need an additional way to specify a key type.

Reply:
@llorllale your case assumes the key value is in the `*did.Doc`, there's a condition in `Create()` that checks if the didDoc.VerificationMethod[] is empty, it will then create a new key from the kms, this is where the key type comes in handy, to create a new key for that keyType option. I will remote that option and use WithOption instead.

Reply:
Yes, I'm aware of the condition. I'm still confused by this interface.

At this moment, `did:key` is the only method I can think of where it makes more sense for the framework to automatically set the verification methods on the resulting `did.Doc`. This doesn't apply to any other method. So we are adding an option that creates ambiguity on the interface, all to support a single DID method.

This option isn't needed - that's why it was never added. The verification methods are being manually set with `did.Doc`.

Reply:
Speaking of which - I think the `key.KeyManager` should also be removed from the method args in the future.

Reply:
We should provide a uniform way to create DID documents - this PR is taking us down an even more fractured path.

Reply:
are you saying this DID:key creator doesn't need to handle kms key creation? (ie VerificaionMethod[] will never be empty?) I'm happy to remove this condition and kms test cases, it will make my life a lot easier!

Reply:
No - `did:key` does not need to automatically create keys. They should be provided to it like all other methods.

Reply:
in case where DID:Key creation doesn't have a full DID document (with empty `[]verificationMethod`) the `Create()` function now accepts a general VDR option (using `WithOption()`) registered as an option key `KeyType` to create a new KMS key and a new `*VerificationMethod` entry to be added to the DIDDoc.

If the option is not passed, then the original behaviour with a DID document having a non empty `[]VerificationMethod` is required.

The custom `WithKeyType()` option has been removed.